From 182e8dd70b8c2414d78f4d1ee8211c8a14149b00 Mon Sep 17 00:00:00 2001
From: Tianshu Qiu <tian.shu.qiu@intel.com>
Date: Wed, 13 Jun 2018 10:58:41 +0800
Subject: [PATCH 087/129] media: intel-ipu4: Fix subdev entity function type

Message for Open Source:
The entity function for subdev is initialized as
MEDIA_ENT_F_V4L2_SUBDEV_UNKNOWN in v4l2_subdev_init.
Should not change to MEDIA_ENT_F_IO_V4L which makes
media-ctl not able to print the pad formats.

Message for Internal:
[Issue/Feature] Refer to open source message.
[Root Cause/Changes]
Remove the assignment for subdev entity function.

Change-Id: I7144ab5e3d3380e1575187f9d7d965e3f9dc0e04
Tracked-On: #JICCCN-173
Signed-off-by: Tianshu Qiu <tian.shu.qiu@intel.com>
Signed-off-by: Meng Wei <wei.meng@intel.com>
---
 drivers/media/pci/intel/ipu-isys-subdev.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/drivers/media/pci/intel/ipu-isys-subdev.c b/drivers/media/pci/intel/ipu-isys-subdev.c
index 8b34cb7..7d81ecc 100644
--- a/drivers/media/pci/intel/ipu-isys-subdev.c
+++ b/drivers/media/pci/intel/ipu-isys-subdev.c
@@ -881,9 +881,6 @@ int ipu_isys_subdev_init(struct ipu_isys_subdev *asd,
 
 	v4l2_subdev_init(&asd->sd, ops);
 
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 5, 0)
-	asd->sd.entity.function = MEDIA_ENT_F_IO_V4L;
-#endif
 	asd->sd.flags |= V4L2_SUBDEV_FL_HAS_DEVNODE | sd_flags;
 	asd->sd.owner = THIS_MODULE;
 
-- 
1.9.1

